merkle_zeppelin/trees/binary.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from functools import cached_property
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class BinaryTree(ABC, Generic[T]):
+    _nodes: list[T]
+    _leafs_number: int
```

Comment:
so leaves instead of leafs... thanks! I didn't know that (somehow 0.o)
regarding `_leaves_number` vs `_number_of_leaves` - I will stay with `_leaves_number`